test_extras/test_chemprop/test_chemprop_pipeline.py

```diff
@@ -306,3 +340,49 @@ def test_prediction(self) -> None:
 
         self.assertEqual(proba.shape, proba_copy.shape)
         self.assertTrue(np.allclose(proba[~nan_indices], proba_copy[~nan_indices]))
+
+
+class TestMulticlassClassificationPipeline(unittest.TestCase):
+    """Test the Chemprop model pipeline for multiclass classification."""
+
+    def test_prediction(self) -> None:
+        """Test the prediction of the multiclass classification model."""
+
+        test_data_df = pd.read_csv(
+            TEST_DATA_DIR / "multiclass_mock.tsv", sep="\t", index_col=False
+        )
+        classification_model = get_multiclass_classification_pipeline(n_classes=3)
+        mols = test_data_df["Molecule"].tolist()
+        classification_model.fit(
+            mols,
+            test_data_df["Label"].to_numpy(),
+        )
+        pred = classification_model.predict(mols)
+        proba = classification_model.predict_proba(mols)
+        self.assertEqual(len(pred), len(test_data_df))
+        self.assertEqual(proba.shape[1], 3)
+        self.assertEqual(proba.shape[0], len(test_data_df))
+
+        model_copy = joblib_dump_load(classification_model)
+        pred_copy = model_copy.predict(mols)
+        proba_copy = model_copy.predict_proba(mols)
+
+        nan_mask = np.isnan(pred)
+        self.assertListEqual(nan_mask.tolist(), np.isnan(pred_copy).tolist())
+        self.assertTrue(np.allclose(pred[~nan_mask], pred_copy[~nan_mask]))
+
+        self.assertEqual(proba.shape, proba_copy.shape)
+        self.assertEqual(pred.shape, pred_copy.shape)
+        self.assertTrue(np.allclose(proba[~nan_mask], proba_copy[~nan_mask]))
+
+        with self.assertRaises(ValueError):
+            classification_model.fit(
+                mols,
+                test_data_df["Label"].add(1).to_numpy(),
+            )
+        with self.assertRaises(ValueError):
+            classification_model = get_multiclass_classification_pipeline(n_classes=2)
+            classification_model.fit(
+                mols,
+                test_data_df["Label"].to_numpy(),
```

Comment:
Two small tests for the `set_params` method should also be added. One test should use valid input and the other should test invalid input that triggers `self._is_valid_multiclass_classifier()` method.